### PR TITLE
[move][clippy] Apply clippy CI check to all targets 

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

- cargo move-clippy did not include tests 

## Test plan 

- CI

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: external-crates/move/crates/bytecode-verifier-tests/src/unit_tests/bounds_tests.rs [hunk 1]
```diff
@@ -222,16 +222,14 @@ fn invalid_signature_for_vector_operation() {
 
     let skeleton = basic_test_module();
     let sig_index = SignatureIndex(skeleton.signatures.len() as u16);
-    for bytecode in vec![
-        VecPack(sig_index, 0),
+    for bytecode in [VecPack(sig_index, 0),
```
Comment (tzakian) [inline external-crates/move/crates/bytecode-verifier-tests/src/unit_tests/bounds_tests.rs:225]: ick is this what cargo fmt does for this? 🤢

File: external-crates/move/crates/move-analyzer/src/symbols.rs [hunk 2]
```diff
@@ -3807,11 +3807,10 @@ fn assert_use_def_with_doc_string(
         "for use in column {use_col} of line {use_line} in file {use_file}"
     );
     let info = def_info.get(&use_def.def_loc).unwrap();
+    let info_str = format!("{}", info);
```
Comment (tzakian) [inline external-crates/move/crates/move-analyzer/src/symbols.rs:3810]: Nit
```suggestion
    let info_str = info.to_string();
```

**Review by tzakian — APPROVED** [review]
LGTM, couple small nits. What cargo fmt did in one place made me sad :'( but nothing to do about it.

File: external-crates/move/crates/move-analyzer/src/symbols.rs (hunk 2 above)
Comment (tnowacki) [reply external-crates/move/crates/move-analyzer/src/symbols.rs:3810]: I'm not sure it implements to_string? Or does anything that has Display has to_string?

File: external-crates/move/crates/bytecode-verifier-tests/src/unit_tests/bounds_tests.rs (hunk 1 above)
Comment (tnowacki) [reply external-crates/move/crates/bytecode-verifier-tests/src/unit_tests/bounds_tests.rs:225]: Looks like it :(

Comment (tnowacki) [reply external-crates/move/crates/bytecode-verifier-tests/src/unit_tests/bounds_tests.rs:225]: Actually it doesn't. Our rustfmt is really fucked for external-crates

File: external-crates/move/crates/move-analyzer/src/symbols.rs (hunk 2 above)
Comment (tzakian) [reply external-crates/move/crates/move-analyzer/src/symbols.rs]: Correct -- anything that is `Display` has `to_string` -- `to_string` calls `Display` underneath it https://doc.rust-lang.org/std/string/trait.ToString.html

**Pull request merged**